spec/lib/soapy_cake/response_value_spec.rb

```diff
@@ -0,0 +1,51 @@
+RSpec.describe SoapyCake::ResponseValue do
+  def subject(*args)
+    described_class.new(*args)
+  end
+
+  describe '#parse' do
+    context 'booleans' do
+      it 'converts the string "true" to the true value' do
+        expect(subject(:foo, 'true').parse).to eq(true)
+      end
+
+      it 'converts the string "false" to the false value' do
+        expect(subject(:foo, 'false').parse).to eq(false)
+      end
+    end
+
+    context 'dates' do
+      it 'parses an ISO-formatted date' do
+        expect(subject(:foo, '2014-06-30T01:00:00').parse).to eq(DateTime.new(2014, 6, 30, 1))
+      end
+    end
+
+    context 'strings' do
+      it 'parses a string' do
+        expect(subject(:foo, 'abc').parse).to eq('abc')
+      end
+    end
+
+    context 'IDs' do
+      it 'converts keys ending in "_id" to integers' do
+        expect(subject(:conversion_id, '42').parse).to eq(42)
+      end
+
+      it "doesn't convert ids in the blacklist to integers" do
+        expect(subject(:tax_id, '123abc').parse).to eq('123abc')
+        expect(subject(:other_tax_id, '123abc').parse).to eq('123abc')
+        expect(subject(:transaction_id, '123abc').parse).to eq('123abc')
+      end
+
+      it 'raises an error if not in the blacklist and non-digit characters are detected' do
+        expect { subject(:foo_id, '123abc').parse }.to raise_error(SoapyCake::Error)
+      end
+
+      it 'does not raise an error if the value is absent' do
```

Comment:
Does this look comprehensive enough? I'm a little worried it may raise an error on some other null value.